### ISSUE TITLE
Google Android Health Connect

### ISSUE THREAD

**Issue by cagnulein** [cagnulein/qdomyos-zwift#1613]
Let's read HR from any WearOS devices

https://developer.android.com/guide/health-and-fitness/health-connect/get-started

**Comment by cagnulein** [comment]
only android 9 or higher is supported, let's go with the google fit one instead

**Comment by cagnulein** [comment]
I guess it's better creating a WearOS app https://github.com/android/health-samples/tree/main/health-services/ExerciseSample

**Comment by fiddur** [comment]
> only android 9 or higher is supported, let's go with the google fit one instead

Google fit is being shut down.  https://www.techadvisor.com/article/2335263/confirmed-google-fit-is-cancelled.html

**Comment by fiddur** [comment]
Are there many installs on Android older than 9 of this app? 

I'm surprised to see Oreo it as high as 3.1% here: https://www.appbrain.com/stats/top-android-sdk-versions

**Comment by cagnulein** [comment]
Yes a lot because a lot of users use qz on old spare phones

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 13 feb 2025 alle 20:33 Fredrik Liljegren <
***@***.***> ha scritto:

**Comment by fiddur** [comment]
Then I suppose you will need different builds for android, since they use different base lib versions, unless that's doable in runtime but I doubt it (but my android dev skills are not much...)

**Comment by cagnulein** [comment]
no it's not doable on runtime. google doesn't allow this. i will think about it.

**Comment by samogot** [comment]
I am interested health connect integration because I want to synchronize exercise statistics into Google Fit instead of Strawa. I'm not very experienced in android, but I though there was a way to specify minimum/target platform version separately and check what the actual version is in runtime and use available API

I might be willing to contribute here

**Comment by cagnulein** [comment]
Yes but the issue is that this will require to close the face down to a lot of qz users that they are using qz on their old android devices.

I will create a separate qz version for this

**Comment by cagnulein** [comment]
done in the version 2.21.5
